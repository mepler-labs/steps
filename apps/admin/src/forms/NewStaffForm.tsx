import Button from 'atoms/Buttons/Button';
import Textarea from 'components/Form/Textarea';
import { Flex } from 'grid-styled';
import * as React from 'react';
import { Form } from 'react-final-form';
import styled from 'styled-components';
import { remCalc } from 'styles/type';

interface Props {
  onSubmit(data): void;
}

export const NewStaffForm = ({ onSubmit }: Props) => {
  return (
    <Form
      onSubmit={onSubmit}
      render={({ handleSubmit }) => (
        <StyledForm onSubmit={handleSubmit}>
          <Flex justifyContent="center">
            <h1>Invite Staff</h1>
          </Flex>
          <p>
            Enter the emails of the people you would like to invite, use commas
            to invite multiple people.
          </p>
          <BaseInputRow>
            <Textarea label="Emails" name="emails" />
          </BaseInputRow>
          <Flex justifyContent="center">
            <Button>Save</Button>
          </Flex>
        </StyledForm>
      )}
    />
  );
};

const StyledForm = styled.form`
  width: ${remCalc(520)};
`;

const BaseInputRow = styled.div`
  textarea {
    margin-bottom: ${remCalc(20)};
    margin-top: ${remCalc(5)};
  }
`;

export default NewStaffForm;
