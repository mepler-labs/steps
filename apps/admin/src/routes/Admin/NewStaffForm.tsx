import * as React from 'react';
import styled from 'styled-components';
import { remCalc } from 'styles/type';
import { reduxForm } from 'redux-form';
import Button from 'atoms/Button';

const BaseInputRow = styled.div`
  display: flex;
  flex-direction: column;
  margin-bottom: ${remCalc(20)};

  label {
    margin-bottom: ${remCalc(5)};
    text-transform: uppercase;
  }

  textarea {
    font-size: ${remCalc(18)};
    padding-bottom: ${remCalc(21)};
    padding-left: ${remCalc(20)};
    padding-right: ${remCalc(20)};
    padding-top: ${remCalc(21)};
    width: 100%;
  }
`;

interface Props {
  onSubmit(data): void;
}

export const NewStaffForm = ({ onSubmit }: Props) => {
  return (
    <form onSubmit={onSubmit}>
      <BaseInputRow>
        <label htmlFor="emails">Emails</label>
        <textarea name="emails" cols={30} rows={10} />
      </BaseInputRow>
      <Button>Save</Button>
    </form>
  );
};

export default NewStaffForm;
