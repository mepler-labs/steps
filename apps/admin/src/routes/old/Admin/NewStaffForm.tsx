import * as React from 'react';
import styled from 'styled-components';
import { remCalc } from 'styles/type';
import Button from 'atoms/Buttons/Button';
import { Form, Field } from 'react-final-form';

interface Props {
  onSubmit(data): void;
}

export const NewStaffForm = ({ onSubmit }: Props) => {
  return (
    <Form
      onSubmit={onSubmit}
      render={({ handleSubmit, pristine, invalid }) => (
        <form onSubmit={handleSubmit}>
          <h1>Invite Staff</h1>
          <p>
            Enter the emails of the people you would like to invite, use commas
            to invite multiple people.
          </p>
          <BaseInputRow>
            <label htmlFor="emails">Emails</label>
            <Field name="emails" component="textarea" />
          </BaseInputRow>
          <Button>Save</Button>
        </form>
      )}
    />
  );
};

const BaseInputRow = styled.div`
  display: flex;
  flex-direction: column;
  margin-bottom: ${remCalc(20)};

  label {
    margin-bottom: ${remCalc(5)};
    text-transform: uppercase;
  }

  textarea {
    font-size: ${remCalc(18)};
    padding-bottom: ${remCalc(21)};
    padding-left: ${remCalc(20)};
    padding-right: ${remCalc(20)};
    padding-top: ${remCalc(21)};
    width: 100%;
  }
`;

export default NewStaffForm;
