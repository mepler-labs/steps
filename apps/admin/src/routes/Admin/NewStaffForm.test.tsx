import { shallow } from 'enzyme';
import 'jest';
import * as React from 'react';
import { NewStaffForm } from './NewStaffForm';

describe('NewStaff.tsx', () => {
  it('renders correctly', () => {
    const onSubmit = jest.fn();
    const wrapper = shallow(<NewStaffForm onSubmit={onSubmit} />);

    expect(wrapper).toMatchSnapshot();
  });
});
