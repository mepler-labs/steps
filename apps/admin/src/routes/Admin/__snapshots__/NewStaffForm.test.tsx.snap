// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewStaff.tsx renders correctly 1`] = `
<ReactFinalForm
  onSubmit={[MockFunction]}
  render={[Function]}
/>
`;
